### PR TITLE
Parametric Imaging update, example fixes and general bug fixing

### LINKED ISSUE #101: Add method to make ASCII image from Image
Should be simple.

https://levelup.gitconnected.com/how-to-convert-an-image-to-ascii-art-with-python-in-5-steps-efbac8996d5e

### LINKED ISSUE #113: Examples need to be updated
I've run all the examples currently on main and a number of them need updating to reflect recent changes. In addition, the vast majority need more documentation (a doc string at the top, an explanation of what each bit is doing etc.)

The reasons for errors are all fairly simple and I've already fixed a bunch but here is the list of outstanding issues:

## Checklist
- [ ] examples/cloudy/run_cloudy.py
- [ ] examples/cloudy/analyse_cloudy.py
- [ ] examples/grids/yggdrasil_example.py
- [ ] examples/grids/plot_spectra_age.py
- [ ] examples/grids/plot_spectra_cloudy.py
- [ ] examples/cosmo/camels.py
- [ ] examples/cosmo/flares.py
- [ ] examples/particle/create_sampled_sed.py
- [ ] examples/particle/create_flux_image_multifilter.py
- [x] examples/parametric/make_image.py
- [x] examples/parametric/generate_observed_sed.py
- [x] examples/parametric/generate_lines.py

### LINKED ISSUE #166: ParametricGalaxy create_image helper method requires updating
Currently doesn't leverage the full flexibility of the underlying `ParametricImage` class.

Example in `examples/parametric/generate_composite_galaxy.py`

### LINKED ISSUE #179: Image uses pixel_values
`Image` uses `pixel_values` but that is specific to `ParticleImage` (I think) and breaks Image functionality for parametric images.

### LINKED ISSUE #194: Updates to `Image` base class initialisation arguments break `ParametricImage`
Arguments here:

https://github.com/flaresimulations/synthesizer/blob/ba481efe28f7eefea872a6fb7bcec3353fea4b2c/synthesizer/imaging/images.py#L1069

Need to be updated with requirements here:

https://github.com/flaresimulations/synthesizer/blob/ba481efe28f7eefea872a6fb7bcec3353fea4b2c/synthesizer/imaging/images.py#L72

### PR DESCRIPTION
**Pull request by WillJRoper**
This PR mainly brings all the Parametric imaging functionality in line with the Particle equivalents. In doing so it's streamlined the morphology class, fixed existing parametric examples, and improved image addition to carry over basic image properties from added images, regardless of image type. 

It also improves the parametric imaging docs and moves them to join the particle_imaging docs while also fixing the other parametric docs.

It additionally fixes the bulk of all examples to work with the new spectra methods and fixes some long-standing failing examples. 

Against better judgment, it implements the ability to print Images in ASCII...

Note that every single example has been rerun on this branch and those that fail fail for reasons outside of this PRs control and will be addressed in an incoming PR.

This PR: closes #101, closes  #113 (with some further fixing to do in another PR), closes  #166, closes  #179, closes  #194 

## Issue Type

- [x] Bug
- [x] Document
- [x] Enhancement

## Checklist

- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by WillJRoper** [comment]
Ah, this also renames `synthesizer.galaxy` -> `synthesizer.base_galaxy` for clarity.

File: synthesizer/sed.py [hunk 1]
```diff
@@ -261,7 +261,7 @@ def get_broadband_fluxes(self, fc, verbose=True):  # broad band flux/nJy
                            'the same as the SED wavelength grid.'))
 
             # Calculate and store the broadband flux in this filter 
-            bb_flux = f.apply_filter(self._lnu, nu=self.nuz) * nJy
+            bb_flux = f.apply_filter(self._fnu, nu=self.nuz) * nJy
```
Comment (christopherlovell) [inline synthesizer/sed.py:264]: this feels like an unrelated change, were broadband luminosities not correctly calculated previously?

**Review by christopherlovell — APPROVED** [review]
